lib/plugins/aws/lib/get-create-change-set-params.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+module.exports = {
+  getCreateChangeSetParams({ changeSetType, templateUrl, templateBody }) {
+    let stackTags = { STAGE: this.provider.getStage() };
+    const stackName = this.provider.naming.getStackName();
+    const changeSetName = this.provider.naming.getStackChangeSetName();
+
+    // Merge additional stack tags
+    if (this.serverless.service.provider.stackTags) {
+      const customKeys = Object.keys(this.serverless.service.provider.stackTags);
+      const collisions = Object.keys(stackTags).filter((defaultKey) =>
+        customKeys.some((key) => defaultKey.toLowerCase() === key.toLowerCase())
+      );
+
+      // Delete collisions upfront
+      for (const key of collisions) {
+        delete stackTags[key];
+      }
+
+      stackTags = Object.assign(stackTags, this.serverless.service.provider.stackTags);
+    }
+
+    const createChangeSetParams = {
+      StackName: stackName,
+      ChangeSetName: changeSetName,
+      ChangeSetType: changeSetType,
+      Capabilities: ['CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM'],
+      Parameters: [],
+      Tags: Object.keys(stackTags).map((key) => ({ Key: key, Value: stackTags[key] })),
+    };
+
+    if (templateUrl) {
+      createChangeSetParams.TemplateURL = templateUrl;
+    }
+
+    if (templateBody) {
+      createChangeSetParams.TemplateBody = JSON.stringify(templateBody);
+    }
+
+    if (
+      (templateUrl &&
+        this.serverless.service.provider.compiledCloudFormationTemplate &&
+        this.serverless.service.provider.compiledCloudFormationTemplate.Transform) ||
+      (templateBody && templateBody.Transform)
+    ) {
+      createChangeSetParams.Capabilities.push('CAPABILITY_AUTO_EXPAND');
+    }
+
+    const customDeploymentRole = this.provider.getCustomDeploymentRole();
+    if (customDeploymentRole) {
+      createChangeSetParams.RoleARN = customDeploymentRole;
+    }
+
+    if (this.serverless.service.provider.notificationArns) {
+      createChangeSetParams.NotificationARNs = this.serverless.service.provider.notificationArns;
+    }
+
+    if (this.serverless.service.provider.stackParameters) {
+      createChangeSetParams.Parameters = this.serverless.service.provider.stackParameters;
+    }
+
+    // Settings below are only applied for `update` type
+    if (changeSetType === 'UPDATE') {
```

Comment:
I'm wondering if we need that distinction at all - do you know why the settings below were never applied to `createStack` previously @medikoo ?

Reply:
I assume it was harmless to apply it only at an update, and doing it also at create, would increase maintenance cost back then.

Otherwise, I assume it's harmless, and I think it can be unconditional

Maybe it even should be unconditional (e.g. if that policy can influence bucket deployment, it's good to have that surface early when creating stack), so maybe let's remove that condition (?)

Reply:
I think it was an ommision previously because I cannot find a good reason why these settings were not supported on `create` in the first place. I've removed that conditional here